src/components/Route.tsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react'
+
+const Route = ({ path, component: Component, ...props }) => {
```

Comment:
Clean solution as well! If there was a lot of routes I'd say we'd want a single controller component that has one navigate listener instead of multiple listeners. But this is fine 👍